R/read_h5ad.R

```diff
@@ -60,13 +60,18 @@ read_h5ad <- function(
     as <- to
   }
 
-  adata <- HDF5AnnData$new(path, mode = mode)
+  hdf5_adata <- HDF5AnnData$new(path, mode = mode)
 
-  switch(
-    as,
-    "SingleCellExperiment" = adata$as_SingleCellExperiment(...),
-    "Seurat" = adata$as_Seurat(...),
-    "InMemoryAnnData" = adata$as_InMemoryAnnData(...),
-    "HDF5AnnData" = adata
-  )
+  if (as == "HDF5AnnData") {
+    hdf5_adata
+  } else {
+    adata <- switch(
+      as,
+      "SingleCellExperiment" = hdf5_adata$as_SingleCellExperiment(...),
+      "Seurat" = hdf5_adata$as_Seurat(...),
+      "InMemoryAnnData" = hdf5_adata$as_InMemoryAnnData(...)
+    )
+    hdf5_adata$close()
+    adata
+  }
```

Comment:
I know, potato potato, but I like this better :3

```suggestion
  if (as == "HDF5AnnData") {
    return(hdf5_adata)
  }
  
  adata <- switch(
    as,
    "SingleCellExperiment" = hdf5_adata$as_SingleCellExperiment(...),
    "Seurat" = hdf5_adata$as_Seurat(...),
    "InMemoryAnnData" = hdf5_adata$as_InMemoryAnnData(...)
  )
  hdf5_adata$close()
  adata
```